code/game/machinery/doors/airlock.dm

```diff
@@ -132,10 +132,13 @@
 		switch(outcome)
 			if(1 to 9)
 				var/turf/here = get_turf(src)
+				var/newtype = /turf/closed/wall
 				for(var/turf/closed/T in range(2, src))
-					here.ChangeTurf(T.type)
-					return INITIALIZE_HINT_QDEL
-				here.ChangeTurf(/turf/closed/wall)
+					newtype = T.type
+					break
```

Comment:
use locate()

Reply:
and to keep on par with robustin's airlock tweak only qdel it if the locate succeeds